### PR TITLE
[Refactor] Updates assets using Swift Protobuf API 

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?

Refactoring assets update methods to use the Swift Protobuf API.

### CODE REVIEW

File: Source/Model/Message/ZMAssetClientMessage+GenericMessage.swift [hunk 1]
```diff
@@ -60,6 +64,17 @@ extension ZMAssetClientMessage {
         return self.cachedGenericAssetMessage
     }
     
+    public var underlyingAssetMessage: GenericMessage? {
```
Comment (typfel) [inline Source/Model/Message/ZMAssetClientMessage+GenericMessage.swift]: We can rename this to `underlyingMessage`, we separated the two in the past for historical reasons.

File: Source/Utilis/Protos/GenericMessage+Helper.swift [hunk 2]
```diff
@@ -401,3 +416,22 @@ public extension LinkPreview {
         }
     }
 }
+
+extension GenericMessage {
+    
+    public mutating func updatedPreview(withAssetId assetId: String, token: String?) {
+        self.asset.preview.remote.assetID = assetId
```
Comment (typfel) [inline Source/Utilis/Protos/GenericMessage+Helper.swift]: Don't we need to handle the ephemeral case here as well?

**Comment by typfel** [comment]
I would run the sync-engine and request-strategy tests with these changes.

**Comment by KaterinaWire** [comment]
> I would run the sync-engine and request-strategy tests with these changes.

There are many failed tests in the sync-engine.

**Comment by KaterinaWire** [comment]
@typfel All tests in SE and request-strategy passed.

**Comment by typfel** [comment]
@KaterinaWire I think we should add some tests for `updatedPreview(withAssetId:token)` and `updatedUploaded(withAssetId:token:)`.

**Comment by KaterinaWire** [comment]
> @KaterinaWire I think we should add some tests for `updatedPreview(withAssetId:token)` and `updatedUploaded(withAssetId:token:)`.

Done

**Comment by KaterinaWire** [comment]
@zenkins release patch

**Pull request merged**